vispy/geometry/generation.py

```diff
@@ -89,6 +89,250 @@ def create_cube():
     return vertices, filled, outline
 
 
+def create_plane(width=1,
+                 height=1,
+                 width_segments=1,
+                 height_segments=1,
+                 direction='+z'):
+    """ Generate vertices & indices for a filled and outlined plane.
+
+    Parameters
+    ----------
+    width : numeric
+        Plane width.
+    height : numeric
+        Plane height.
+    width_segments : int
+        Plane segments count along the width.
+    height_segments : numeric
+        Plane segments count along the height.
+    direction: unicode
+        ``{'-x', '+x', '-y', '+y', '-z', '+z'}``
+        Direction the plane will be facing.
+
+    Returns
+    -------
+    vertices : array
+        Array of vertices suitable for use as a VertexBuffer.
+    faces : array
+        Indices to use to produce a filled plane.
+    outline : array
+        Indices to use to produce an outline of the plane.
+
+    References
+    ----------
+    [1] Cabello, R. (n.d.). PlaneBufferGeometry.js. Retrieved May 12, 2015,
+        from https://github.com/mrdoob/three.js/blob/master/src/extras/geometries/PlaneBufferGeometry.js  #noqa
+    """
+
+    x_grid = width_segments
+    y_grid = height_segments
+
+    x_grid1 = x_grid + 1
+    y_grid1 = y_grid + 1
+
+    # Positions, normals and texcoords.
+    positions = np.zeros(x_grid1 * y_grid1 * 3)
+    normals = np.zeros(x_grid1 * y_grid1 * 3)
+    texcoords = np.zeros(x_grid1 * y_grid1 * 2)
+
+    offset = offset1 = 0
+    for i_y in range(y_grid1):
```

Comment:
I think these loops can be cut down a bit:
```
y = np.arange(y_grid1) * height / y_grid - height / 2
x = np.arange(x_grid1) * width / x_grid - width / 2

positions[::3] = np.tile(x, y_grid1)
positions[1::3] = - np.repeat(y, x_grid1)

normals[2::3] = 1

texcoords[::2] = np.tile(np.arange(x_grid1) / x_grid, y_grid1)
texcoords[1::2] = np.repeat(1 - np.arange(y_grid1) / y_grid, x_grid1)
```
(I may have messed up a few of the limits.)